src/csc.rs

```diff
@@ -134,6 +134,24 @@ impl<'a> CscMatrix<'a> {
             });
         }
     }
+
+    /// Constructs a matrix from an iterator
+    ///
+    /// The iterator should deliver values in column major format of a dense data block, otherwise 
+    /// there are more efficient methods to construct the sparse matrix. Note that this copies all
+    /// elements of the matrix.
+    ///
+    pub fn from_iterator_dense<'b, I: IntoIterator<Item=&'b float>>(nrows: usize, ncols: usize, iter: I) -> CscMatrix<'static> {
+        use std::iter;
+
+        CscMatrix {
+            nrows,
+            ncols,
+            indptr: (0..ncols+1).map(|i| i * nrows).collect::<Vec<usize>>().into(),
+            indices: iter::repeat(0..nrows).take(ncols).flatten().collect::<Vec<usize>>().into(),
+            data: iter.into_iter().cloned().collect::<Vec<float>>().into(),
```

Comment:
Could you check that `data.len() == nrows * ncols` and panic if that invariant is not upheld.